crates/edr_napi/test/solidity-tests.ts

```diff
@@ -0,0 +1,58 @@
+import chai, { assert } from "chai";
+import chaiAsPromised from "chai-as-promised";
+
+import { SolidityTestRunner, TestSuite, TestContract, ArtifactId, SuiteResult } from "..";
+
+describe("Solidity Tests", () => {
+  it("executes basic tests", async function() {
+    const testSuites = [loadContract("./SetupConsistencyCheck.json"), loadContract("./PaymentFailureTest.json")];
+
+    const resultsFromCallback: Array<SuiteResult> = [];
+    const runner = new SolidityTestRunner((...args) => {
+      resultsFromCallback.push(args[1] as SuiteResult);
+    });
+
+    const result = await runner.runTests(testSuites);
+    assert.equal(resultsFromCallback.length, testSuites.length);
+    assert.equal(resultsFromCallback.length, result.length);
+
+    for (let res of result) {
```

Comment:
Is there a simpler way to validate results other than doing a name comparison string?

Are the results guaranteed to be in the same order as the inputs?

Reply:
Nope, they'll be in the order of completion. Forge returns them as a BTreeMap, but can't expose that through NAPI afaik, so I opted for a list of `(name, value)` pairs to preserve the semantics. If it's not important though to preserve the order of completion, we could just return a map of `name => value` which would make it easy to query for specific test suites. cc @fvictorio

Reply:
As a caller of the API, I think it would be more ergonomic if the order of the array of results was in the same order as the input array. That way you can easily iterate them side-by-side and there is no need for unwieldy code and expensive name comparisons.

I had a look at the code and it seems possible to achieve this by using an integer ID to assign a result to an index of a pre-allocated array matching the length of the input array. Alternatively, a safer alternative might be to use oneshot channels per test suite. That way you could `.collect` all results of those oneshot channels.

Reply:
Thanks good points, but thinking about this more I think it's good to return from Rust as `(name, value)` pairs preserving the order of completion for maximum flexibility, as it's easy to turn that into any other desired format in JS (since JS knows the input order). Otoh JS can also know the order of completion from the callbacks, so maybe it's not important. Would like to hear @fvictorio 's feedback who is experimenting from the JS side.